### DISCUSSION TITLE
How to force the use of asyncpg?

### DISCUSSION THREAD

**Discussion by ghost** [sqlalchemy/sqlalchemy#10015] [Usage Questions] [answered]
I want to use asyncpg even though the url is like
```
postgresql://postgres:postgres@localhost:5432
```

**Comment by CaselIT** [comment]
Hi

You will need to modify the url object to set the wanted driver before creating the engine

**Reply by ghost** [reply]
Are you referring to the url string as above or some specific url object?

**Reply by CaselIT** [reply]
The url object obtained by https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.engine.make_url

But modifying the string works too

**Reply by ghost** [reply, marked as answer]
Thanks. make_url and then URL.set is a good way.
